tidy3d/plugins/resonance/resonance.py

```diff
@@ -0,0 +1,272 @@
+        eigvecs: np.ndarray[complex],
+    ) -> np.ndarray[complex]:
+        """Find the eigenvalue error"""
+        J = len(eigvals)
+        errors = np.zeros(J)
+        for k in range(J):
+            errors[k] = np.linalg.norm(
+                (u_matrices[2] - (eigvals[k] ** 2) * u_matrices[0]) @ eigvecs[:, k]
+            )
+        return errors
+
+    def _iterate(
+        self, signal: np.ndarray[complex], prev_eigvals: np.ndarray[complex]
+    ) -> ResonanceData:
+        """Run a single iteration of the solver."""
+        u_matrices = self._evaluate_matrices(signal, prev_eigvals)
+
+        eigvals, eigvecs = self._solve_gen_eig_prob(u_matrices[1], u_matrices[0], self.rcond)
+
+        errors = self._find_errors(eigvals, u_matrices, eigvecs)
+
+        amplitudes = self._find_amplitudes(signal, prev_eigvals, eigvals, eigvecs)
+
+        return ResonanceData(eigvals=eigvals, errors=errors, complex_amplitudes=amplitudes)
+
+    def run(self, signal: np.ndarray[complex], time_step: float) -> xr.Dataset:
+        """Finds resonances in a time series.
+        Note that the signal should start after the sources have turned off.
+
+        Parameters
+        ----------
+        signal : np.ndarray[complex]
+            Time series to search for resonances
+        time_step : float
+            Time step / sampling rate of the data (in seconds).
+
+        Returns
+        -------
+        :class:`.xr.Dataset`
+            Dataset containing the frequency, decay rate, Q, amplitude, phase,
+            and estimation error of the resonances. Modes with low Q, small
+            amplitude, or high estimation error are likely to be spurious.
+        """
+        J = self.freq_grid
+        if self.freq_grid is None:
+            J = int(time_step * len(signal) * (self.freq_window[1] - self.freq_window[0]) / 2)
+        if J == 0:
+            raise SetupError("Attempting to start with no grid frequencies.")
+        omegas = np.linspace(
+            time_step * self.freq_window[0] * 2 * np.pi,
+            time_step * self.freq_window[1] * 2 * np.pi,
+            J,
+        )
+        eigvals = np.exp(-1j * omegas)
+        prev_num_eigvals = -1
+        for _ in range(1 + 2 * J):
+            resdata = self._iterate(signal, eigvals)
+            if len(resdata.eigvals) == prev_num_eigvals:
+                break
+            prev_num_eigvals = len(resdata.eigvals)
+
+        resonances = self._get_resonance_info(data=resdata, time_step=time_step)
+
+        return resonances.sortby("freq")
+
+    def run_field_time(self, signal: ScalarFieldTimeDataArray) -> xr.Dataset:
+        """Finds resonances in a ScalarFieldTimeDataArray.
+        Note that the signal should start after the sources have turned off."""
+        dts = np.diff(signal.t)
+        dt = dts[0]
+        if not np.allclose(dts, dt, rtol=1e-5, atol=0):
+            raise SetupError("The time step must be constant.")
+        return self.run(signal.data, dt)
```

Comment:
Any thoughts on providing another helper method to process a `FieldTimeData` directly?
```py
def run_field_time(field_time: FieldTimeData) -> xr.Dataset
```
would basically call the `run_scalar_field_time` method on each of the field components in the `FieldTimeData`?